### ISSUE TITLE
Error: Type mismatch in argument 'destructor' at (1); passed INTEGER(8) to INTEGER(4)

### ISSUE THREAD

**Issue by barracuda156** [interkosmos/fortran-sqlite3#2]
```
src/sqlite3.f90:734:32:

  734 |             sqlite3_bind_text = sqlite3_bind_text_(stmt, idx, val, len(val), destructor)
      |                                1
Error: Type mismatch in argument 'destructor' at (1); passed INTEGER(8) to INTEGER(4)
make: *** [libfortran-sqlite3.a] Error 1
```

Passing ` -fallow-argument-mismatch` did not help.

**Comment by interkosmos** [comment]
Do you compile on a 32-bit platform?

**Comment by barracuda156** [comment]
> Do you compile on a 32-bit platform?

Yes, this is on `ppc`.

**Comment by interkosmos** [comment]
Should be fixed with commit 33b1f0da566cd4aaa36e59ceb70be8c89a6a3443. Thanks for reporting.

**Comment by barracuda156** [comment]
@interkosmos Thank you, the build succeeds now.

Test output:
```
SQLite library version: 3.41.2
SQLite source ID: 2023-03-22 11:56:21 0d1fc92f94cb6b76bffe3ec34d69cffde2924203304e8ffc4155597af0c191da
DB name: main
----------------------------------------------------------------
ERROR 1: TEST LOG
----------------------------------------------------------------
Turning WAL mode on ...
SQLite version from query: 3.41.2
Row 1 has been added to table test_table in database main!
Row 2 has been added to table test_table in database main!
Row 3 has been added to table test_table in database main!

--- TESTING PREPARE/STEP
           1         one       12345
           2         two   987654321
           3       three   192837465

--- TESTING CALLBACK FUNCTION
--- There are 3 values in selected row
VALUE: 1
VALUE: one
VALUE: 12345
--- There are 3 values in selected row
VALUE: 2
VALUE: two
VALUE: 987654321
--- There are 3 values in selected row
VALUE: 3
VALUE: three
VALUE: 192837465
```
